apps/ejabberd/rebar.config.script

```diff
@@ -21,15 +11,49 @@ GetFullVer = fun() ->
     end
 end,
 
-case erlang:system_info(otp_release) of
-    "17" ->
-        case catch GetFullVer() of
-            <<"17.5">> -> % This is 17.5 where cover was fixed
-                CONFIG;
-            _ ->
-                io:format("disabling coverage for OTP 17.0 to 17.4~n"),
-                DisableCoverage()
-        end;
-    _ ->
-        CONFIG
-end.
+
+DisableCoverage = fun(Config) ->
+    Opts = [{cover_enabled, false},
+            {cover_print_enabled, false},
+            {cover_export_enabled, false}],
+    lists:foldl(fun({Key, _}=Tuple, Acc) ->
+                        lists:keyreplace(Key, 1, Acc, Tuple)
+                end, Config, Opts)
+end,
+
+MaybeDisableCoverage = fun(Config) ->
+    case erlang:system_info(otp_release) of
+        "17" ->
+           case catch GetFullVer() of
+                <<"17.5">> -> % This is 17.5 where cover was fixed
+                    Config;
+                _ ->
+                    io:format("disabling coverage for OTP 17.0 to 17.4~n"),
+                    DisableCoverage(Config)
+           end;
+        _ ->
+           Config
+    end
+end,
+
+MaybeChangeOpenSSL = fun(Config) ->
+    case os:getenv("OPENSSL_LIB") of
+        false ->
+            Config;
+        OpenSSL ->
+            io:format("using OpenSSL ~p~n", [OpenSSL]),
```

Comment:
This clause body is on a different level of abstraction, so might use exporting to a well named function.